components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java

```diff
@@ -403,9 +403,21 @@ private int getDeploymentStatus(String nodeURL, String siddhiAppName, int busine
             }
             return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
         } catch (SiddhiAppsApiHelperException e) {
-            if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED) {
-                return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
+           if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED && e.getStatus() == 404) {
+               return TemplateManagerConstants.SIDDHI_APP_DEPLOYED;
+           } else  if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_DEPLOYED) {
+               return TemplateManagerConstants.SIDDHI_APP_UNREACHABLE;
+           }
+           if (businessRuleStatus == TemplateManagerConstants.PARTIALLY_UNDEPLOYED && e.getStatus() == 404) {
+               return TemplateManagerConstants.SIDDHI_APP_NOT_DEPLOYED;
```

Comment:
Why this is SIDDHI_APP_NOT_DEPLOYED when there is a possibility of still deployed in the unreachable node(s).

Reply:
changed in the code

Reply:
@thushaaanthan please put the commit in a comment for the fixes.